scanners/generic/generic_podman.py

```diff
@@ -0,0 +1,185 @@
+    # PROTECTED CONSTANTS                                         #
+    # Accessed by parent generic object                               #
+    ###############################################################
+
+    def __init__(self, config, ident="generic"):
+        """Initialize all vars based on the config.
+        The code of the function only deals with the "podman" layer, the "generic" layer is handled by super()
+        """
+
+        logging.debug("Initializing podman-based generic scanner")
+        # Initialize generic scanner
+        super().__init__(config, ident)
+
+        image = self.my_conf("container.parameters.image")
+        if not image:
+            logging.error(
+                f"Generic podman scanner requires an image to load (`scanners.{self.ident}.container.parameters.image`)"
+            )
+
+        # Initialize podman
+        self.podman = PodmanWrapper(
+            app_name=self.config.get("application.shortName"),
+            scan_name=self.ident,
+            image=image,
+        )
+
+    ###############################################################
+    # PUBLIC METHODS                                              #
+    # Accessed by RapiDAST                                        #
+    # + MUST be implemented                                       #
+    # + SHOULD call super().<method>                              #
+    # + list: setup(), run(), postprocess(), cleanup()            #
+    ###############################################################
+
+    def setup(self):
+        """Prepares everything:
+        - the command line to run
+        - environment variables
+        - files & directory
+
+        The code of the function only deals with the "podman" layer, the "generic" layer is handled by super()
+        """
+
+        if self.state != State.UNCONFIGURED:
+            raise RuntimeError(
+                f"generic_podman setup encountered an unexpected state: {self.state}"
+            )
+
+        self._setup_podman_cli()
+        self._setup_generic_cli()
+
+        super().setup()
+
+        if self.state != State.ERROR:
+            self.state = State.READY
+
+    def run(self):
+        """If the state is READY, run the final podman run command.
+        There is no need to call super() here.
+        """
+        logging.info("Running up the generic scanner in podman")
+        if not self.state == State.READY:
+            raise RuntimeError("[generic SCANNER]: ERROR, not ready to run")
+
+        cli = self.podman.get_complete_cli(self.generic_cli)
+
+        # The result is stdout if "container.results" is undefined or `*stdout`
+        result_is_stdout = (
+            not self.my_conf("container.results")
+            or self.my_conf("container.results") == "*stdout"
+        )
+
+        # DO STUFF
+        logging.info(f"Running generic with the following command:\n{cli}")
+        scanning = subprocess.run(
```

Comment:
updated with the following change:
- just for sake of performance optimization, the pipe is created only when required (i.e.: `result == '*stdout' or `None``)
- when storing output, I use `print(, end="")` instead of `logging` to have a clean output (without meddling about), in case it's important, to avoid causing alignment issues.

However, I'm pretty sure it can be further optimized to avoid having to store in the `scanning_stdout_results` buffer.

Let me know if that looks okay